hcloud/action.go

```diff
@@ -141,14 +141,21 @@ func (c *ActionClient) All(ctx context.Context) ([]*Action, error) {
 
 // WatchProgress watches the actions progress until it completes with success or error.
 func (c *ActionClient) WatchProgress(ctx context.Context, action *Action) (<-chan int, <-chan error) {
+
+	return c.WatchProgressMS(ctx, action, 500)
+}
+
+// WatchProgressMS watches the actions progress until it completes with success or error. Makes it possible to configuration the polling timeout.
+func (c *ActionClient) WatchProgressMS(ctx context.Context, action *Action, timeoutMs int) (<-chan int, <-chan error) {
```

Comment:
Third parameter should be `timeout time.Duration`. Then I would suggest to rename the method to `WatchProgressTimeout`.